### PR TITLE
Re-authoring the UAP package to make sure we have the right set of assemblies

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard @ericstj 

FYI: @tijoytom 

Fixes to the UAP package so that we have the right set of assemblies for ref and lib required to actually consume the package.

### CODE REVIEW

File: src/System.DirectoryServices.Protocols/src/Configurations.props [hunk 1]
```diff
@@ -2,7 +2,8 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <BuildConfigurations>
-        netstandard-Windows_NT;
+    	netcoreapp-Windows_NT;
```
Comment (weshaggard) [inline src/System.DirectoryServices.Protocols/src/Configurations.props]: nit: indention.

File: src/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj [hunk 2]
```diff
@@ -34,7 +34,7 @@
     <Reference Include="System.Diagnostics.Debug" />
     <Reference Include="System.Diagnostics.Tools" />
     <Reference Include="System.IO.FileSystem" />
-    <Reference Include="System.IO.FileSystem.AccessControl" />
+    <Reference Include="System.IO.FileSystem.AccessControl" Condition="'$(TargetGroup)' != 'uap'" />
```
Comment (weshaggard) [inline src/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj]: I assume this is going to break the build.

File: src/System.Text.Encoding.CodePages/dir.props [hunk 3]
```diff
@@ -3,5 +3,6 @@
   <Import Project="..\dir.props" />
   <PropertyGroup>
     <AssemblyVersion>4.1.0.0</AssemblyVersion>
+    <IsUAP>true</IsUAP>
```
Comment (weshaggard) [inline src/System.Text.Encoding.CodePages/dir.props]: Why do we need this in UAP? What depends on it?

File: src/shims/dir.props [hunk 4]
```diff
@@ -15,14 +15,18 @@
     <NetFxReference Include="System" />
     <NetFxReference Include="System.Core" />
     <NetFxReference Include="System.ComponentModel.Composition" />
+    <NetFxReference Include="System.ComponentModel.DataAnnotations" />
     <NetFxReference Include="System.Data" />
     <NetFxReference Include="System.Drawing" />
     <NetFxReference Include="System.IO.Compression.FileSystem" />
+    <NetFxReference Include="System.Net" />
```
Comment (weshaggard) [inline src/shims/dir.props]: We should look at this as I don't think it has any APIs. However we might have the type-forwards but we should double check.

File: src/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj (hunk 2 above)
Comment (joperezr) [inline src/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj]: no, we have different implementation for uap, so this is actually not used any longer in uap

File: src/System.DirectoryServices.Protocols/src/Configurations.props (hunk 1 above)
Comment (joperezr) [inline src/System.DirectoryServices.Protocols/src/Configurations.props]: will fix

File: src/System.Text.Encoding.CodePages/dir.props (hunk 3 above)
Comment (joperezr) [inline src/System.Text.Encoding.CodePages/dir.props]: It was in the previous shipped uap package

Comment (vitek-karas) [inline src/System.Text.Encoding.CodePages/dir.props]: The .NET Native toolchain has a dependency on this assembly. We detect if the app needs encodings which are not normally supported and add a dependency to this assembly in that case.

Comment (tijoytom) [inline src/System.Text.Encoding.CodePages/dir.props]: Yes , There are apps which use non-standard encoding.

Comment (weshaggard) [inline src/System.Text.Encoding.CodePages/dir.props]: While I don't like it, I'm OK with it.

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Release x64 Build and Test please

**Pull request merged**